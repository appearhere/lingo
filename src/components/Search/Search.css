.root {
  justify-self: end;
  grid-area: search;
  overflow: hidden;
  min-width: 0;
  min-height: 0;
  width: 100%;
}

.inputRoot {
  width: 100%;
}

input[type="text"].input {
  padding-top: var(--size-regular);
  padding-bottom: var(--size-regular);
  background-color: var(--color-greyLightest);
}

input[type="text"].input:focus {
  outline: none;
  background-color: var(--color-greyLightest);
  border-color: var(--color-greyLighter);
}
