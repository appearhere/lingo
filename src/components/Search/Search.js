import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router';

import IconInput from '@appearhere/bloom/components/Form/IconInput/IconInput';

import css from './Search.css';

class Search extends Component {
  static propTypes = {
    match: PropTypes.shape({}),
    location: PropTypes.shape({}),
    history: PropTypes.shape({}),
  };

  handleInputChange = (e) => {
    const { value } = e.target;

    this.updateURL(value);
  };

  checkLocation = () => {
    const { match, history } = this.props;

    if (match.params.department) history.push('/');
  };

  updateURL = (query) => {
    const { location, history } = this.props;

    if (query) {
      history.push(`${location.pathname}?q=${query}`);
    } else {
      history.push(`${location.pathname}`);
    }
  };

  render() {
    const { location } = this.props;
    const value = location.search.substr(3);

    return (
      // eslint-disable-next-line
      <div className={ css.root } onClick={ this.checkLocation }>
        <IconInput
          iconName="search"
          iconSide="right"
          type="search"
          classNames={ { root: css.inputRoot, input: css.input } }
          placeholder="Search..."
          onChange={ this.handleInputChange }
          value={ value }
        />
      </div>
    );
  }
}

export default withRouter(Search);
