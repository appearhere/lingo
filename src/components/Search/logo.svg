<?xml version="1.0" encoding="UTF-8"?>
<svg width="25px" height="22px" viewBox="0 0 25 22" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: Sketch 48.2 (47327) - http://www.bohemiancoding.com/sketch -->
    <title>Combined Shape</title>
    <desc>Created with Sketch.</desc>
    <defs></defs>
    <g id="-Search" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd" transform="translate(-13.000000, -14.000000)">
        <path d="M13,14 L18.9608064,14 L18.9608064,16.5944518 L16.2059804,16.5944518 L16.2059804,32.5561654 L18.9608064,32.5561654 L18.9608064,35.1517394 L13,35.1517394 L13,14 Z M31.3280805,35.1517394 L31.3280805,32.5561654 L34.0829066,32.5561654 L34.0829066,16.5944518 L31.3280805,16.5944518 L31.3280805,14 L37.2900009,14 L37.2900009,35.1517394 L31.3280805,35.1517394 Z" id="Combined-Shape" fill="#FFFFFF"></path>
    </g>
</svg>