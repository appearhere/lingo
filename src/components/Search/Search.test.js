import React from 'react';
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router';

import Search from './Search';

describe('Search component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<StaticRouter><Search /></StaticRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
