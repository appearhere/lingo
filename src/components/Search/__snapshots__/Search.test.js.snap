// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component renders correctly 1`] = `
<div
  className={undefined}
  onClick={[Function]}
>
  <div
    className=""
  >
    <span
      className=""
      dangerouslySetInnerHTML={
        Object {
          "__html": "icon-search.svg",
        }
      }
      onClick={[Function]}
    />
    <div
      className={undefined}
    >
      <input
        className=""
        id={undefined}
        name={undefined}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Search..."
        required={undefined}
        type="text"
        value=""
      />
      <span
        className={undefined}
      />
    </div>
  </div>
</div>
`;
